### PR TITLE
Enable CODEOWNERS for iOS/macOS/Darwin directories.

### PR DESCRIPTION
**Pull request by julianrex**
This may be short-lived, but still could be useful - and I think it's worth an experiment. I've kept it deliberately light too.

I didn't assume to add an Android/Qt/etc sections - but feel free to add.

/cc @chloekraw @1ec5

### CODE REVIEW

**Comment by chloekraw** [comment]
I’m all for iteration and experimentation, so sounds good to me!

**Comment by friedbunny** [comment]
[`CODEOWNERS` docs](https://help.github.com/articles/about-codeowners/) — as it stands, this PR will request (but not require) reviews from the listed groups.

File: .github/CODEOWNERS [hunk 1]
```diff
@@ -0,0 +1,9 @@
+# iOS
+/platform/ios/ @maps-ios-reviewers
```
Comment (friedbunny) [inline .github/CODEOWNERS]: I believe these groups need to be qualified with the organization, e.g., `@mapbox/maps-ios-reviewers`.

/cc @mapbox/maps-ios-reviewers 😛

File: .github/CODEOWNERS [hunk 2]
```diff
@@ -0,0 +1,9 @@
+# iOS
+/platform/ios/ @mapbox/maps-ios-reviewers
+
+# macOS
+/platform/macos/ @1ec5 @mapbox/maps-ios-reviewers
+
+# Darwin
+/platform/darwin/ @mapbox/maps-ios-reviewers
```
Comment (1ec5) [inline .github/CODEOWNERS]: platform/darwin/ affects macOS just as much as iOS, so I might want to be notified for this path, or maybe I should join the maps-ios-reviewers team. 🤷‍♂️

Comment (julianrex) [reply .github/CODEOWNERS]: Happy to do either - was trying to avoid extra notifications for you 😉

**Pull request merged**